onnx_tf/handlers/backend/erf.py

```diff
@@ -0,0 +1,25 @@
+import tensorflow as tf
+
+from onnx_tf.handlers.backend_handler import BackendHandler
+from onnx_tf.handlers.handler import onnx_op
+from onnx_tf.handlers.handler import tf_func
+
+
+@onnx_op("Erf")
+class Erf(BackendHandler):
+
+  @classmethod
+  def version_9(cls, node, **kwargs):
+    inp = kwargs["tensor_dict"][node.inputs[0]]
+    dtype = inp.dtype
+    if dtype != tf.half and dtype != tf.float32 and dtype != tf.float64:
```

Comment:
Better to assert and fail and silently do something users do not expect.
@chinhuang007 @fumihwh what do you think?

Reply:
This is a general question, do we try to 1. silently cast between ONNX and TF whenever there is a type mismatch? or 2. assert and fail in these situations, indicating certain data types are not supported in specific TF operators? Approach 2 seems safer, though more restrictive because some ONNX models will fail to convert to TF.

Reply:
For me I prefer 1 because of the accountability issue. If we take actions to deviate from user-specified semantics (even though they may appear obviously wrong) and if downstream task fails in any significant ways due to our deviations, we are the ones to blame whereas for approach 2, we are always not accountable because we always follow users' instructions. So to conclude, I'd go for 1.

Ideally, we can put automatic type conversion under non-strict mode. (c.f. the strict flag here: https://github.com/onnx/onnx-tensorflow/blob/master/doc/CLI.md)

Reply:
So I believe you suggest we assert and fail with error message when the data type is not explicitly supported by TF. As an option, we could add code to attempt auto-conversion only under non-strict mode. However based on the current CLI doc, "Changing strict mode to False is strongly discouraged"

Reply:
Yes, I believe we should assert and fail. And I think the suggestion for not using strict mode is quite accurate. Silently changing the semantics of the model shifts the responsibility of failure, if there is any, to **us** instead of the model developer.